contracts/BaseLaunchpeg.sol

```diff
@@ -397,43 +325,20 @@ abstract contract BaseLaunchpeg is
         emit PublicSaleEndTimeSet(_publicSaleEndTime);
     }
 
-    /// @notice Set the reveal batch size. Can only be set after
-    /// batch reveal has been initialized and before a batch has
-    /// been revealed.
-    /// @dev Only callable by owner
-    /// @param _revealBatchSize New reveal batch size
-    function setRevealBatchSize(uint256 _revealBatchSize)
+    /// @notice Set the withdraw AVAX start time.
+    /// @param _withdrawAVAXStartTime New public sale end time
+    function setWithdrawAVAXStartTime(uint256 _withdrawAVAXStartTime)
         external
         override
         onlyOwner
     {
-        _setRevealBatchSize(_revealBatchSize);
+        withdrawAVAXStartTime = _withdrawAVAXStartTime;
+        emit WithdrawAVAXStartTimeSet(_withdrawAVAXStartTime);
     }
 
-    /// @notice Set the batch reveal start time. Can only be set after
-    /// batch reveal has been initialized and before a batch has
-    /// been revealed.
-    /// @dev Only callable by owner
-    /// @param _revealStartTime New batch reveal start time in seconds
-    function setRevealStartTime(uint256 _revealStartTime)
-        external
-        override
-        onlyOwner
-    {
-        _setRevealStartTime(_revealStartTime);
-    }
-
-    /// @notice Set the batch reveal interval. Can only be set after
-    /// batch reveal has been initialized and before a batch has
-    /// been revealed.
-    /// @dev Only callable by owner
-    /// @param _revealInterval New batch reveal interval in seconds
-    function setRevealInterval(uint256 _revealInterval)
-        external
-        override
-        onlyOwner
-    {
-        _setRevealInterval(_revealInterval);
+    /// @notice Update batch reveal
+    function setBatchReveal(address _batchReveal) external override onlyOwner {
```

Comment:
should we check that `IBatchReveal(_batchReveal).baseLaunchpeg() == address(this)`? will we also allow setting zero addresses to disable the batch reveal? in this case, would it be cleaner to not use `revealBatchSize` of `0` to indicate that the batch reveal is disabled (and to disallow that value from being set)?

Reply:
Yeah good point we should only allow to set a batch reveal contract initialized for the given launchpeg: https://github.com/traderjoe-xyz/launchpeg/pull/38/commits/16d2ce41c560aa04c1f081b35a513f5e3b4c5f92

For your other point, yes we can disable the batch reveal if we don't set it (this is probably a bit cleaner than the other approach right?)
I've made some changes here, let me know if I'm missing anything : https://github.com/traderjoe-xyz/launchpeg/pull/38/commits/b4eb150b4a25e1243fef9428d8d0ea5376977ddf

Reply:
yup i think it looks good. 

some questions:
- do we want to allow the owner to set the `BatchReveal` contract to a zero address again after it has been set previously?
- i think the current tokenURI logic will return the URI as if there is no BatchReveal when the contract is first initialized, instead of the unrevealed URI. is there any issue with that?

Reply:
1. yup i'm fine allowing it to be set to `address(0)` again. i think we might need to update the `setBatchReveal(..)` method to handle that.
2. in the current `tokenURI()` logic we return the `baseURI` if there is no BatchReveal contract. but this is also the case when we first deploy the contract (prior to calling `initializePhases(...)`)

```
if (address(batchReveal) == address(0)) {
    return string(abi.encodePacked(baseURI, _id.toString()));
} 
```